src/coreclr/vm/jitinterface.cpp

```diff
@@ -11798,7 +11798,17 @@ InfoAccessType CEEJitInfo::constructStringLiteral(CORINFO_MODULE_HANDLE scopeHnd
     }
     else
     {
-        *ppValue = (LPVOID)ConstructStringLiteral(scopeHnd, metaTok); // throws
+        if (!GetModule(scopeHnd)->IsCollectible() && !IsCompilingForNGen() && !IsReadyToRunCompilation())
```

Comment:
```suggestion
        if (!GetModule(scopeHnd)->IsCollectible())
```
This method is never called for those cases.

Reply:
It may be a good idea to do a similar change for R2R. The change for that would be in crossgen2.

Reply:
I am wondering if it makes sense for R2R, since R2R code cannot access the string literal directly anyway, we probably cannot save in terms of code size.

Reply:
We can save one indirection (ie one instruction) for R2R.